### PR TITLE
Cleaning & killing some material-ui

### PR DESCRIPTION
**Pull request by rasmusguterstam**
## What?
- Add `List` component to `hedvig-ui`
- Add `Dropdown` component to `hedvig-ui`
- Make `Dropdown` and `EnumDropdown` automatically show loading indicator if their `onChange` is async
- Add `Shadowed` to `@hedvig-ui/typography`
- Replace a lot (not all) of `material-ui` components with `hedvig-ui` equivalents
- Remove some good 'ol `Formik` forms
- Don’t render claim payment table if it is empty, but show appropriate message
- Update the “Contract for Claim” dropdown under Claim Information

## Why?
- We're trying to move away from `material-ui` and instead use our own `hedvig-ui` components to (1) facilitate and speed up development of new features; (2) provide consistency to Hope. It should be easy to make things pretty, intuitive and efficient for the users of Hope.
- Hope is kind of slow (not as bad as it was before though), and we're aiming to reduce the number of heavy packages (where `material-ui` currently is our main target)

### Claim Information

Before             |  After
:-------------------------:|:-------------------------:
 | 

### Payments

Before
image

After
image

## Optional checklist
- [ ] Updated `src/changelog.ts`
- [x] Codescouted
- [ ] Unit tests written
- [x] Tested locally

### CODE REVIEW

File: shared/hedvig-ui/dropdown.tsx [hunk 1]
```diff
@@ -23,17 +89,23 @@ export const EnumDropdown: React.FC<{
...
   return (
     <StyledDropdown
+      value={value}
       placeholder={placeholder}
       options={dropdownOptions}
+      loading={loading ?? autoLoading}
       selection
-      onChange={(_, { value }) => setValue(value)}
+      onChange={async (_, { value: selection }) => {
+        setAutoLoading(true)
```
Comment (joacimastrom) [inline shared/hedvig-ui/dropdown.tsx:105]: I feel like we may be over-engineering this component with the auto-loader and allowing options as `option[] | string[]` and then mapping them

It's often more maintainable to stay as close to native behaviour as possible and keep single responsibility, so the component is used the same way everywhere :)

**Review by joacimastrom — APPROVED** [review]
All in all great cleanup! 🌟 

Some minor comments, I don't feel too strongly about any of them so it's your call ;) Feel free to discuss anything with me

File: shared/hedvig-ui/dropdown.tsx [hunk 2]
```diff
@@ -1,19 +1,85 @@
...
+        }
+      }
+
+      return option
+    }),
+  ]
+
+  return (
+    <StyledDropdown
+      icon={getOptions().length === 0 ? null : undefined}
```
[the remark this answers is not in the document]
Comment (rasmusguterstam) [reply shared/hedvig-ui/dropdown.tsx:54]: It's a very silly way of letting semantic-ui default to using the dropdown arrow (`undefined`) or, in the case of only one item in the dropdown, don't render an icon at all (`null`). Perhaps there's a better way to do this? 😬

File: shared/hedvig-ui/dropdown.tsx [hunk 3]
```diff
@@ -1,19 +1,85 @@
...
+  value,
+  loading,
+  onRender,
+  emptyLabel,
+  className,
+  ...props
+}) => {
+  const [autoLoading, setAutoLoading] = useState(false)
+
+  const getOptions = (): DropdownValue[] => [
```
[the remark this answers is not in the document]
Comment (rasmusguterstam) [reply shared/hedvig-ui/dropdown.tsx:38]: I always forget that `useMemo` exists haha, but this is the day I'm using it! 😎

File: shared/hedvig-ui/dropdown.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (joacimastrom) [reply shared/hedvig-ui/dropdown.tsx:54]: Haha okay, but can there be a selected if there are no options? 🤔  I guess this is fine 😇  though I wouldn't mind if the arrow always was there even if there are no options

File: shared/hedvig-ui/dropdown.tsx (hunk 1 above)
Comment (rasmusguterstam) [reply shared/hedvig-ui/dropdown.tsx]: Really good point!

**Pull request merged**